tests/everflow/test_everflow_testbed.py

```diff
@@ -61,27 +61,38 @@ class EverflowIPv4Tests(BaseEverflowTest):
 
     DEFAULT_SRC_IP = "20.0.0.1"
     DEFAULT_DST_IP = "30.0.0.1"
-
-    @pytest.fixture(params=["tor", "spine"])
-    def dest_port_type(self, request):
+    
+    @pytest.fixture
+    def setup_dest_port_type(self, request):
         """
-        Get the dest port for this test case.
-
-        Used to parametrize test cases based on dest port type to validate forwarding behavior.
+        This fixture get the dest port type and can perform action based
+        on that. As of now cleanup is being done here.
 
         Args:
             request: Pytest request object
 
         Returns:
             A destination port type in {tor, spine}.
         """
-        return request.param
+        yield request.param
+        
+        duthosts = request.getfixturevalue("duthosts")
+        rand_one_dut_hostname = request.getfixturevalue("rand_one_dut_hostname")
+        setup_info = request.getfixturevalue("setup_info")
+        setup_mirror_session =  request.getfixturevalue("setup_mirror_session")
+        tbinfo = request.getfixturevalue("tbinfo")
+
+        duthost = duthosts[rand_one_dut_hostname]
+
+        for index in range(0, min(3, len(setup_info[request.param]["dest_port"]))):
+            tx_port = setup_info[request.param]["dest_port"][index]
+            peer_ip, _ = everflow_utils.get_neighbor_info(duthost, tx_port, tbinfo)
+            everflow_utils.remove_route(duthost, setup_mirror_session["session_prefixes"][0], peer_ip)
+            everflow_utils.remove_route(duthost, setup_mirror_session["session_prefixes"][1], peer_ip)
 
-    # FIXME: We use a lot of try-catch here to clean up routes, but it's a bit messy.
-    # A better solution might be to have a fixture that receives route updates and then
-    # cleans up any remaining routes at the end.
 
-    def test_everflow_basic_forwarding(self, duthosts, rand_one_dut_hostname, setup_info, setup_mirror_session, dest_port_type, ptfadapter, tbinfo):
+    @pytest.mark.parametrize('setup_dest_port_type', ["tor", "spine"], indirect=['setup_dest_port_type'])
```

Comment:
I'm still not very clear on why this is needed/what the difference is between this and just passing a `param` kwarg to the fixture?

Reply:
@daall Updated not to specify indirect. I am using the model referred here:https://stackoverflow.com/questions/18011902/pass-a-parameter-to-a-fixture-function/60148972#60148972.

Basically i need to pass dest_port_type and need to take some action based on that before test execution starts itself.

Reply:
I think that model makes sense, but what is the difference between taking action based on this `dest_port_type` param vs. `request.param`?

Reply:
@daall I think i got your point. Let me try that where we need not add mark parameterize on each test case and use request.param directly

Reply:
@daall Thanks it works fine. Updated accordingly.